src/entities/trades/0x-trade.ts

```diff
@@ -0,0 +1,272 @@
+import invariant from 'tiny-invariant'
+
+import { ChainId, ONE, TradeType } from '../../constants'
+import { Currency } from '../currency'
+import { CurrencyAmount } from '../fractions/currencyAmount'
+import { Fraction } from '../fractions/fraction'
+import { Percent } from '../fractions/percent'
+import { Price } from '../fractions/price'
+import { TokenAmount } from '../fractions/tokenAmount'
+import { currencyEquals, Token } from '../token'
+import { Trade } from './interfaces/trade'
+import { UnsignedTransaction } from '@ethersproject/transactions'
+import { Breakdown, Platform } from '../platforms-breakdown'
+import { RoutablePlatform } from './routable-platform/routable-platform'
+import fetch from 'node-fetch'
+import { BigNumber } from '@ethersproject/bignumber'
+import { Decimal } from 'decimal.js-light'
+
+interface ApiSource {
+  name: string
+  proportion: string
+}
+
+interface ApiResponse {
+  price: string
+  guaranteedPrice: string
+  to: string
+  data: string
+  value: string
+  gas: string
+  estimatedGas: string
+  gasPrice: string
+  protocolFee: string
+  minimumProtocolFee: string
+  buyTokenAddress: string
+  sellTokenAddress: string
+  buyAmount: string
+  sellAmount: string
+  estimatedGasTokenRefund: string
+  sources: ApiSource[]
+}
+
+const decodePlatformName = (apiName: string): string => {
+  switch (apiName) {
+    case 'Uniswap_V2':
+      return 'Uniswap v2'
+    case 'Liquidity provider':
+      return 'LP'
+    case 'Balancer_V2':
+      return 'Balancer v2'
+    case 'DODO_V2':
+      return 'Dodo v2'
+    case 'Uniswap_V3':
+      return 'Uniswap v3'
+    case 'PancakeSwap_V2': // shouldn't be used since it's on BSC, but added to be extra sure
+      return 'PancakeSwap v2'
+    default:
+      return apiName
+  }
+}
```

Comment:
```suggestion

/**
 * A map to get the platform name from its code
 */
export const CODE_TO_PLATFORM_NAME_MAP = {
  Uniswap_V2: 'Uniswap v2',
  'Liquidity provider': 'LP',
  Balancer_V2: 'Balancer v2',
  DODO_V2: 'Dodo v2',
  Uniswap_V3: 'Uniswap v3',
  PancakeSwap_V2: 'PancakeSwap v2', // shouldn't be used since it's on BSC, but added to be extra sure
};

/**
  * Gets the platform from its unique identifier
  * @param {string} The platform ID
  * @returns The platform name if found. Otherwise, the values passed
  */
const decodePlatformName = (apiName: string): string => CODE_TO_PLATFORM_NAME_MAP[apiName] || apiName;
```

Reply:
Ahahah I literally don't know why I didn't use a map :facepalm: